docs/topics/Permissions.md

```diff
@@ -26,39 +26,40 @@ Below is a table of all current permissions, their integer values in hexadecimal
+| MENTION_EVERYONE      | `0x000020000` | Allows for using the `@everyone` tag to notify all users in a channel, and the `@here` tag to notify all online users in a channel | T            |
+| USE_EXTERNAL_EMOJIS   | `0x000040000` | Allows the usage of custom emojis from other servers                                                                               | T            |
+| VIEW_GUILD_INSIGHTS   | `0x000080000` | Allows for viewing guild insights                                                                                                  |              |
+| CONNECT               | `0x000100000` | Allows for joining of a voice channel                                                                                              | V, S         |
+| SPEAK                 | `0x000200000` | Allows for speaking in a voice channel                                                                                             | V            |
+| MUTE_MEMBERS          | `0x000400000` | Allows for muting members in a voice channel                                                                                       | V, S         |
+| DEAFEN_MEMBERS        | `0x000800000` | Allows for deafening of members in a voice channel                                                                                 | V, S         |
+| MOVE_MEMBERS          | `0x001000000` | Allows for moving of members between voice channels                                                                                | V, S         |
+| USE_VAD               | `0x002000000` | Allows for using voice-activity-detection in a voice channel                                                                       | V, S         |
+| CHANGE_NICKNAME       | `0x004000000` | Allows for modification of own nickname                                                                                            |              |
+| MANAGE_NICKNAMES      | `0x008000000` | Allows for modification of other users nicknames                                                                                   |              |
+| MANAGE_ROLES \*       | `0x010000000` | Allows management and editing of roles                                                                                             | T, V, S      |
+| MANAGE_WEBHOOKS \*    | `0x020000000` | Allows management and editing of webhooks                                                                                          | T            |
+| MANAGE_EMOJIS \*      | `0x040000000` | Allows management and editing of emojis                                                                                            |              |
+| REQUEST_TO_SPEAK      | `0x100000000` | Allows for requesting to speak in stage channels. (_This permission is under active development and may be changed or removed._)   | S            |
```

Comment:
This causes 32-bit int overflow. Even unsigned.

Reply:
which is why they are sent as strings, you should parse them with your language's BigInt (or similar) functionality, or a library for it. permissions should be stored as bigints, read the docs: https://discord.com/developers/docs/topics/permissions